### PR TITLE
Rename texture atlas texture field

### LINKED ISSUE #268: TextureAtlas textures field has confusing name
The [TextureAtlas](https://github.com/bevyengine/bevy/blob/db1bf6478c98d822292415179cb0d69863c8bc5b/crates/bevy_sprite/src/texture_atlas.rs#L13) struct in bevy_sprites has a field named `textures` which is of type `Vec<Rect>`. I think this field should be named `sprites`, since it holds the rectangles of the sprites in the atlas. I also think the function `add_texture` should be renamed to `add_sprite`.

Is there any particular reason the field is called textures instead of sprites?

### PR DESCRIPTION
**Pull request by HPRIOR**
Resolves #268, renaming `textures` field to `sprites` and the method `add_texture` to `add_sprites` in the `TextureAtlas` struct.

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
CI failure is spurious: https://opengameart.org/content/fox-and-shiba was failing.

File: crates/bevy_sprite/src/texture_atlas.rs [hunk 1]
```diff
@@ -129,18 +129,18 @@ impl TextureAtlas {
...
-    pub fn add_texture(&mut self, rect: Rect) -> usize {
-        self.textures.push(rect);
-        self.textures.len() - 1
+    pub fn add_sprite(&mut self, rect: Rect) -> usize {
+        self.sprites.push(rect);
+        self.sprites.len() - 1
     }
 
-    /// How many textures are in the `TextureAtlas`
+    /// How many sprites are in the `TextureAtlas`
```
Comment (alice-i-cecile) [inline crates/bevy_sprite/src/texture_atlas.rs:137]:
```suggestion
    /// How many sprites are in the [`TextureAtlas`]
```

While we're here, might as well linkify things.

File: crates/bevy_sprite/src/texture_atlas.rs [hunk 2]
```diff
@@ -17,7 +17,7 @@ pub struct TextureAtlas {
     // TODO: add support to Uniforms derive to write dimensions and sprites to the same buffer
     pub size: Vec2,
     /// The specific areas of the atlas where each texture can be found
```
Comment (alice-i-cecile) [inline crates/bevy_sprite/src/texture_atlas.rs:19]: This doc comment is now out of date, and could generally be expanded.

**Comment by alice-i-cecile** [comment]
Hey, thanks! Your changes look great, and I only have a few minor nits.

On the topic of the issue itself, I think this makes sense, but I don't really understand why this is called a "texture atlas" in the first place. Given that we're only using it for 2D, wouldn't using the synonymous "sprite sheet" be clearer?

**Comment by HPRIOR** [comment]
Hey thanks for the review. It's my first time contributing to open source so I thought I'd start with something light. 

> It's a bit strange to me to have a `TextureAtlas` contain a `sprites` field, but not a `textures` field.
Yeah that does seem strange. Shall I change `TextureAtlas` to `SpriteSheet`, and also some other associated structs like `TextureAtlasBuilder` and `DynamicTextureAtlasBuilder`, or is this something that needs more discussion? Lots of other variables and comments would need to be changed too. 

> This doc comment is now out of date, and could generally be expanded.
Sorry I'm not sure why this is out of date or how to expand it as I'm new to the code base.

>CI failure is spurious: https://opengameart.org/content/fox-and-shiba was failing.
Is this a flaky test? The checks seem to have passed now

**Comment by alice-i-cecile** [comment]
> Yeah that does seem strange. Shall I change TextureAtlas to SpriteSheet, and also some other associated structs like TextureAtlasBuilder and DynamicTextureAtlasBuilder, or is this something that needs more discussion?

This definitely needs more discussion. I think that bringing it up on Discord in #rendering-dev will be the best way to solicit quick, casual feedback. Overall, I think I'm reluctant to say we should merge this before we have a clear answer there.

> Is this a flaky test? The checks seem to have passed now

Yep, that particular CI test is to check that all of our links are valid. But of course, links are inherently flaky, and sometimes we get random failures like that.

**Comment by superdump** [comment]
I disagree with this change as mentioned just now in the issue. In my mind, a sprite sheet is either just use of a texture atlas for the purpose of sprites, or that plus some special cases for sprites. A texture atlas is the general term and will be used for more things than just sprites. For example, I intend to build a shadow atlas for point lights soon. This will require a texture atlas.

**Pull request closed without merging**